code/android-core-library/src/phone/java/com/adobe/marketing/mobile/services/caching/FileCacheService.java

```diff
@@ -0,0 +1,126 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile.services.caching;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.adobe.marketing.mobile.services.DeviceInforming;
+import com.adobe.marketing.mobile.services.Log;
+import com.adobe.marketing.mobile.services.ServiceConstants;
+import com.adobe.marketing.mobile.util.StringUtils;
+
+import java.io.File;
+import java.util.Date;
+import java.util.Map;
+
+/**
+ * A {@code CacheService} that uses file storage as the cache store.
+ */
+public class FileCacheService implements CacheService {
```

Comment:
Can you move these classes into internal subpackage? We don't want extensions directly creating an instance of this class.